src/test/java/com/training/services/tests/ProductsTest.java

```diff
@@ -158,24 +163,21 @@ public void testShorexOfferingDateValidate() throws ParseException {
      */
     @Test
     public void testShorexProductTypeNameValidate() {
-        String productTypeName;
+        response = new RestEngine().getResponseGet(baseURL, headerMap, queryParam).as(Response.class);
 
-        response = new RestEngine().getResponseGet(baseURL, headerMap, queryParam)
-                .as(Response.class);
         Assertions.assertThat(response.getStatus()).isEqualTo(200).as("Json response status is not 200");
 
         List<Products> products = response.getPayload().getProducts();
 
-        for (int index = 0; index < products.size(); index++) {
-
-            productTypeName = products.get(index).getProductType().getProductTypeName();
+        IntStream.range(0, products.size()).forEach(index -> {
+            String productTypeName = products.get(index).getProductType().getProductTypeName();
 
             softAssert.assertThat((productTypeName.equalsIgnoreCase("Shore Excursion")
                     ||
                     productTypeName.equalsIgnoreCase("Aquatics")))
                     .as("Product Id = " +
                             products.get(index).getProductID() + " product type name is not shore excursion or aquatics")
                     .isTrue();
-        }
+        });
```

Comment:
Do we really need index in this?
Why cant we use - response.getPayload().getProducts().forEach()?